### PR TITLE
Fixed LOAD DATA errors

### PR DESCRIPTION
**Pull request by Hydrocharged**
There are three core issues with `LOAD DATA` that have been fixed:
1) Columns could be rearranged, but we didn't handle the case where the rearranged columns were a subset.
2) Line endings were assumed to be a single character, which is not true for Windows (which uses `\r\n`).
3) Default values did not work. Nil default values "worked" due to nil being the default value when instantiating a slice of `interface{}`s.

### CODE REVIEW

**Comment by timsehn** [comment]
@VinaiRachakonda should review this since he built the initial implementation.

File: enginetest/load_queries.go [hunk 1]
```diff
@@ -74,6 +75,110 @@ var LoadDataScripts = []ScriptTest{
...
+	},
+	{
+		Name: "LOAD DATA handles non-nil default values",
+		SetUpScript: []string{
+			"CREATE TABLE test1 (pk BIGINT PRIMARY KEY, v1 BIGINT DEFAULT (v2 * 10), v2 BIGINT DEFAULT 5);",
+			"CREATE TABLE test2 (pk BIGINT PRIMARY KEY, v1 BIGINT DEFAULT (v2 * 10), v2 BIGINT DEFAULT 5);",
+			"CREATE TABLE test3 (pk BIGINT PRIMARY KEY, v1 BIGINT DEFAULT (pk * 10), v2 BIGINT DEFAULT (v1 - 1));",
+			"CREATE TABLE test4 (pk BIGINT PRIMARY KEY, v1 BIGINT DEFAULT (pk * 10), v2 BIGINT DEFAULT (v1 - 1));",
+			"LOAD DATA INFILE './testdata/test7.txt' INTO TABLE test1;",
+			"LOAD DATA INFILE './testdata/test7.txt' INTO TABLE test2 (pk);", // The (pk) projection results in a different tree
```
Comment (zachmu) [inline enginetest/load_queries.go:116]: Thanks for adding column projections, this is a pretty big gap in coverage.

Could we get at least one test that re-orders the columns from their schema order? I.e. `into table test5 (v2, v1, pk)`

**Review by zachmu — APPROVED** [review]
LGTM, missing one test case

**Review by VinaiRachakonda — APPROVED** [review]
LGTM!

**Pull request merged**